registry/storage/vacuum.go

```diff
@@ -27,14 +27,20 @@ type Vacuum struct {
 	ctx    context.Context
 }
 
-// RemoveBlob removes a blob from the filesystem
-func (v Vacuum) RemoveBlob(dgst string) error {
+// RemoveBlob removes a blob from the filesystem. withDirectory
+// allows to remove <hex-digest> directory containing the blob as well.
+func (v Vacuum) RemoveBlob(dgst string, withDirectory bool) error {
```

Comment:
> This interface addition makes no sense. What would removing a blob ever have to do with a directory? This leaks any notion of abstraction.

You're right. The argument shouldn't be there. Empty directory should be removed implicitly. Or is there any problem with that approach? I wonder why they are kept in all the delete/clear/remove methods in the registry. I smell a reason behind that.

> Also, the type here should be a digest, not a string.

That was my impression as well. But I lacked courage to change that :-).